app/controllers/collection_templates_controller.rb

```diff
@@ -46,6 +52,21 @@ def create
     end
   end
 
+  def update
+    if names_list(params[:id]).include?(params[:draft][:template_name])
+      flash[:error] = 'A template with that name already exists.'
+      redirect_to manage_collections_path
+      return
+    end
+
+    begin
```

Comment:
How many checks for duplicate template names are there? can they be consolidated?

Also, shouldn't our front end validation (if working correctly) also prevent a user from saving/updating if a template name already exists? If that is the case, we probably should still have a just in case check in the controller, but we should definitely consolidate.

Reply:
Currently, the UI, the controller, and DB all check.  The front end validation cannot be trusted because the list of names is created when the page is loaded.  For instance, submitting, going back (browser) and then submitting again circumvents this validation.  The controller one can probably be removed, but I consider the DB one mandatory.

Reply:
Ok, so if the DB check covers the other one in the controller, let's just keep the one. However, the way this is written seems to break rails convention in the interest of keeping the method of the superclass intact. I think we should try to use a model validation, which should catch this error. If that means we need to tweak or rewrite some of the code in the superclass for the method to follow the rails convention then I think that is acceptable.

Reply:
https://guides.rubyonrails.org/v4.2/active_record_validations.html